src/AcceptanceTests/Receiving/When_receiving_a_raw_json_message.cs

```diff
@@ -24,7 +25,9 @@ public async Task ShouldConsumeIt()
                 {
                     b.CustomConfig((cfg, context) =>
                     {
-                        cfg.UseSerialization<NewtonsoftSerializer>();
+                        // Hack to get serializer settings in unwrapper to use TypeNameHandling.Auto
+                        cfg.GetSettings().Set("NServiceBus.Newtonsoft.Json.Settings", new JsonSerializerSettings { TypeNameHandling = TypeNameHandling.Auto });
+                        cfg.UseSerialization<NewtonsoftJsonSerializer>();
```

Comment:
Wouldn't this work: https://docs.particular.net/nservicebus/serialization/newtonsoft#usage-custom-settings

```suggestion
                        var settings = new JsonSerializerSettings
                        {
                            TypeNameHandling = TypeNameHandling.Auto,
                            Converters =
                            {
                                new IsoDateTimeConverter
                                {
                                    DateTimeStyles = DateTimeStyles.RoundtripKind
                                }
                            }
                        };
                        var serialization = endpointConfiguration.UseSerialization<NewtonsoftJsonSerializer>();
                        serialization.Settings(settings);
```